### ISSUE TITLE
Absolute Values of Quantities

### ISSUE THREAD

**Issue by ferittuncer** [angularsen/UnitsNet#325]
We can improve readability, writeability and maybe also performance (by avoiding unnecessary `.Meters` and `.FromMeters`) if we provide conversion to abstract.

As-is: `Length abstractLength1 = Length.FromMeters(Math.Abs(length1.Meters))`

To-be: `Length abstractLength1 = length1.ToAbstract()`

What do you think?

**Comment by JKSnd** [comment]
I hope this doesn't come across as rude, but do you mean "Absolute" instead of "Abstract"?

**Comment by angularsen** [comment]
You probably refer to absolute. Sure, this is probably convenient, but I do question how many or how often this adds value since it's a matter of a bit more syntax to achieve the same thing and it will increase the size of the library linearly with the number of quantities. I'm really worried about the size of the library lately and am hesitant to add anything that is not essential, frequently used or solves something that is truly cumbersome to do outside the library.

I'm starting to think this is actually another argument for a common quantity interface + extension method or an abstract base quantity type, to avoid duplicating the implementation N times. Discussed in #308 .

**Comment by ferittuncer** [comment]
Was working overtime a lot lately. And when I'm tired, I do these kind of silly mistakes :) Yeah it should be absolute value instead of abstract value.

I feel that getting absolute values easily is an essential feature for a quantity library, because the whole point of UnitsNet is to make dealing with quantities easy, right?

But also as you said, if it costs O(n) space, not feasible. If we would have a base abstract class it would be O(1) space.

So I agree that we should deal with #308 first.

**Comment by angularsen** [comment]
I am closing this due to inactivity, please reopen if you plan to pursue this further.

As noted in #302 , we can possibly do also add this to our `IQuantity<T>` interface, but I'm not sure I feel the effort is worth the slight improvement in syntax.

**Comment by AndrewWern** [comment]
> I do question how often this adds value since it's a matter of a bit more syntax to achieve the same thing

Late to the party, but I have written my own absolute value functions in nearly every app I've authored at my new job. The most common use is for getting a delta between two values. An actuator needs to be able to hold position ±0.02 inches, so I check that the absolute value of the difference of the actual position from the commanded position is less than the allowed threshold. Repeat for force, current, speed... It's hard to look at `Length posErr = Length.FromInches(Math.Abs(_fix.actualPosition.Inches - commandPosition.Inches));` so I just made a utility function for each unit I use.
